src/cursor/cursor.ts

```diff
@@ -39,9 +43,58 @@ export type CursorFlag = typeof FLAGS[number];
 /** @public */
 export const FIELDS = ['numberOfRetries', 'tailableRetryInterval'] as const;
 
+/** @public */
+export interface DocumentTransforms {
+  /** Transform each document returned */
+  doc(doc: Document): Document;
+  /** Transform the value returned from the initial query */
+  query?(doc: Document): Document | Document[];
+}
+
+/** @internal */
+export interface CoreCursorPrivate {
+  /** Transforms functions */
+  transforms?: DocumentTransforms;
+  numberOfRetries: number;
+  tailableRetryInterval: number;
+  currentNumberOfRetries: number;
+  explicitlyIgnoreSession: boolean;
+  batchSize: number;
+
+  state: CursorState;
+  readConcern?: ReadConcern;
+}
+
 /** @internal */
 export type CursorPrivate = CoreCursorPrivate;
 
+/** @public Possible states for a cursor */
+export enum CursorState {
```

Comment:
is `CursorState` used outside of this file? I know you didn't write this code, but I think a lot of the curious exporting we did with cursors was due to the fact that we had to split between `core_cursor` and `cursor`

Reply:
Yeah it's used in a number of places current; in the CommandCursor and in several places in the operation layer.